app/lib/audit.js

```diff
@@ -70,6 +77,57 @@ function format_record(r){
     return s;
 }
 
+function validate_record(r, k, ln){
+    var v = false;
+    var checks = 0;
+    var checks_available = [];
+    var passed = 0;
+    var checks_passed = [];
+    if (r){
+        var old_keys = Object.keys(r[0]);
```

Comment:
We should really use "let" rather than "var" - vars aren't scoped the way they're written. (Not worth changing for this.)